.buildkite/publish-react-native-aztec-android-artifacts.sh

```diff
@@ -0,0 +1,10 @@
+#!/bin/bash
+
+set -euo pipefail
```

Comment:
Completely opinionated, but as I mentioned to Gio in the other PR, personally I've seen the practice of setting script-wide bash flags like this directly on the shebang instead of using a separate `set` command. Ultimately both do the same, it's just that I've seen the latter used more often in bash script examples and tutorials so considered it might be more common practice? 🤷 

```suggestion
#!/bin/bash -euo pipefail
```

Reply:
Buildkite examples use the extra `set` approach, so I'd prefer to stick with it at least until there is consensus.